medcat/utils/meta_cat/ml_utils.py

```diff
@@ -412,11 +412,23 @@ def eval_model(model: nn.Module, data: List, config: ConfigMetaCAT, tokenizer: T
     precision, recall, f1, support = precision_recall_fscore_support(y_eval, predictions, average=score_average)
 
     labels = [name for (name, _) in sorted(config.general['category_value2id'].items(), key=lambda x: x[1])]
-    confusion = pd.DataFrame(
-        data=confusion_matrix(y_eval, predictions, ),
-        columns=["true " + label for label in labels],
-        index=["predicted " + label for label in labels],
-    )
+
+    try:
```

Comment:
What is (potentially) raising the ValueError here (the `confusion_matrix` call or the (pd.DataFrame` one)? Perhaps we'd be able to check for incompatibility instead of relying on try-catch? Or at the very least narrow it down a little further. Right now we've duplicated code here for the call to `confusion_matrix` as well as `pd.DataFrame`.

Shouldn't we always be able to use the `labels_present` approach?
And if the length of `labels` and `labels_present` differs, you could still log that as you do now.

Reply:
> Yes, we could do the `labels_present` approach each time as well, and log the warning. Should I make the change?

I think that would be a better approach since it a) limits the duplication and b) avoid catching an exception. Given we're catching quite a broad exception here, we don't really know for certain that what we expect went wrong. A `ValueError` may also be raised for some other reason by the same method(s). And we'd be capturing it here. Yet might not actually know how to handle it.

Reply:
Sure, let me make the change and remove the try except block